crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,501 @@
+                    _view,
+                ) => {
+                    debug!("Got commit and meta {:?}", payload_commitment);
+                    commit_and_metadata = Some(CommitmentAndMetadata {
+                        commitment: *payload_commitment,
+                        metadata: metadata.clone(),
+                    });
+                }
+                HotShotEvent::QCFormed(cert) => match cert {
+                    either::Right(timeout) => {
+                        _timeout_certificate = Some(ViewChangeEvidence::Timeout(timeout.clone()));
+                    }
+                    either::Left(qc) => _quorum_certificate = Some(qc),
+                },
+                HotShotEvent::ViewSyncFinalizeCertificate2Recv(cert) => {
+                    _view_sync_finalize_cert = Some(cert);
+                }
+                _ => {}
+            }
+        }
+
+        if commit_and_metadata.is_none() {
+            error!(
+                "Somehow completed the proposal dependency task without a commitment and metadata"
+            );
+            return;
+        }
+
+        broadcast_event(
+            Arc::new(HotShotEvent::QuorumProposalDependenciesValidated(
+                self.view_number,
+            )),
+            &self.sender,
+        )
+        .await;
+    }
+}
+
+/// The state for the quorum proposal task.
+pub struct QuorumProposalTaskState<TYPES: NodeType, I: NodeImplementation<TYPES>> {
+    /// Latest view number that has been proposed for.
+    pub latest_proposed_view: TYPES::Time,
+
+    /// Table for the in-progress proposal depdencey tasks.
+    pub propose_dependencies: HashMap<TYPES::Time, JoinHandle<()>>,
+
+    /// Network for all nodes
+    pub quorum_network: Arc<I::QuorumNetwork>,
+
+    /// Network for DA committee
+    pub committee_network: Arc<I::CommitteeNetwork>,
+
+    /// Output events to application
+    pub output_event_stream: async_broadcast::Sender<Event<TYPES>>,
+
+    /// Reference to consensus. The replica will require a write lock on this.
+    pub consensus: Arc<RwLock<Consensus<TYPES>>>,
+
+    /// Membership for Timeout votes/certs
+    pub timeout_membership: Arc<TYPES::Membership>,
+
+    /// Membership for Quorum Certs/votes
+    pub quorum_membership: Arc<TYPES::Membership>,
+
+    /// The node's id
+    pub id: u64,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPES, I> {
+    /// Create an event dependency
+    #[instrument(skip_all, fields(id = self.id, latest_proposed_view = *self.latest_proposed_view), name = "Quorum vote create event dependency", level = "error")]
+    fn create_event_dependency(
+        &self,
+        dependency_type: ProposalDependency,
+        view_number: TYPES::Time,
+        event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+    ) -> EventDependency<Arc<HotShotEvent<TYPES>>> {
+        EventDependency::new(
+            event_receiver,
+            Box::new(move |event| {
```

Comment:
So, as I have come to understand this, we return the view number to make sure that all of the events are for the correct view, then they are collected into the `Vec<HotShotEvent>` and sent down the pipe. So I followed Keyao's lead here and just did the same thing she did. Do I need to do anything else?